console/src/main.rs

```diff
@@ -15,6 +15,25 @@ mod tasks;
 mod term;
 mod view;
 
+enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+async fn connect(
+    target: String,
+    should_delay: bool,
```

Comment:
take it or leave it: this essentially indicates whether or not a connection attempt is a reconnect, right? the delay is the behavior that `connect` does when it's reconnecting --- it might, eventually, also do other behavior. personally, i might call this `is_reconnect` or something.

not a blocker though!